src/v/cloud_topics/level_one/metastore/replicated_metastore.cc

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "cloud_topics/level_one/metastore/replicated_metastore.h"
+
+#include "cloud_topics/level_one/metastore/frontend.h"
+#include "cloud_topics/level_one/metastore/rpc_types.h"
+#include "cloud_topics/level_one/metastore/state_update.h"
+#include "cloud_topics/logger.h"
+
+namespace experimental::cloud_topics::l1 {
+
+namespace {
+
+// Convert metastore error codes to RPC error codes
+metastore::errc rpc_to_meta_errc(rpc::errc ec) {
+    switch (ec) {
+    case rpc::errc::ok:
+        dassert(false, "Unexpected OK code");
+        return static_cast<metastore::errc>(0); // Should not be called with ok
```

Comment:
i don't really understand this. if it's not supposed to happen, why not use `vassert(false)` to catch logic errors?

Reply:
I'd originally been conservative thinking that it's not worth a crash, especially since the rpc::errc comes from the wire (which might be corrupted or something). But you're right that this here always indicates a logic error, corruption or not.